### PR TITLE
common/strtol: replace `const char*` with `std::string_view`

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/strtol.cc [hunk 1]
```diff
@@ -237,7 +237,7 @@ uint64_t strict_iecstrtoll(std::string_view str, std::string *err)
 
 uint64_t strict_iecstrtoll(const char *str, std::string *err)
 {
-  return strict_iec_cast<uint64_t>(std::string_view(str), err);
+  return strict_iec_cast<uint64_t>(str, err);
```
Comment (rzarzynski) [inline src/common/strtol.cc]: ACK, no change in logic and overhead.

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -3322,7 +3322,7 @@ int main(int argc, const char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = strict_iec_cast<long long>(val.c_str(), &err);
+      max_size = strict_iec_cast<long long>(val, &err);
```
Comment (rzarzynski) [inline src/rgw/rgw_admin.cc:3325]: Should we squash this commit with the previous one? Is the buildability preserved between them?

**Review by rzarzynski — APPROVED** [review]
LGTM with just a minor question regarding commit atomicity.

File: src/rgw/rgw_admin.cc (hunk 2 above)
Comment (tchaikov) [reply src/rgw/rgw_admin.cc:3325]: indeed. folded.

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-07-26_12:44:08-rados-wip-kefu-testing-2021-07-26-1839-distro-basic-smithi/

**Pull request merged**